### ISSUE TITLE
[data grid] Blank header and rows while mounting

### ISSUE THREAD

**Issue by bromanazzi** [mui/mui-x#6595]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

In the codesandbox below, when switching tabs quickly, sometimes the header and rows delay rendering for a brief period. 

### Expected behavior 🤔

The rows and header should render immediately to prevent a flash of empty grid. The footer already does this.

### Steps to reproduce 🕹

Link to live example: https://codesandbox.io/s/laughing-hopper-0mv8ll

Steps:
1. Click on the tab buttons to switch views
2. You should see a brief flash

### Context 🔦

I'm developing a view with tabs, where users are expected to switch quickly between them to compare information from different grids. However, every time a new grid mounts its blank for a brief period, enough to be disruptive. 

I understand in the codesandbox it can be solved by always rendering the same DataGrid. The Page components are meant to represent larger, different views.

I searched for issues on stackoverflow and on here, but couldn't find anything. Is there a way to opt into some kind of immediate rendering?

### Your environment 🌎

<details>
  <summary><code>npx @mui/envinfo</code></summary>
```
  System:
    OS: Windows 10 10.0.19044
  Binaries:
    Node: 18.8.0 - C:\Program Files\nodejs\node.EXE
    Yarn: Not Found
    npm: 8.18.0 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Chrome: Not Found
    Edge: Spartan (44.19041.1266.0), Chromium (106.0.1370.47)
  npmPackages:
    @emotion/react: ^11.9.0 => 11.10.4
    @emotion/styled: ^11.8.1 => 11.10.4
    @mui/base:  5.0.0-alpha.95
    @mui/core-downloads-tracker:  5.10.3
    @mui/icons-material: ^5.6.2 => 5.10.3
    @mui/lab: ^5.0.0-alpha.98 => 5.0.0-alpha.98
    @mui/material: ^5.6.1 => 5.10.3
    @mui/private-theming:  5.10.3
    @mui/styled-engine:  5.10.4
    @mui/system:  5.10.4
    @mui/types:  7.2.0
    @mui/utils:  5.10.3
    @mui/x-data-grid: ^5.17.7 => 5.17.7
    @mui/x-date-pickers: ^5.0.0-alpha.2 => 5.0.0-beta.7
    @types/react: ^18.0.18 => 18.0.18
    react: ^18.0.0 => 18.2.0
    react-dom: ^18.0.0 => 18.2.0
    typescript: ^4.8.2 => 4.8.2
```
</details>

**Comment by bromanazzi** [comment]
Thank you for the detailed response! 

Changing the visibility of the tab panels instead of unmounting them is something we tried early in the project, but decided against it. It got quite hard to follow lifecycles, and there was a lot of extra code just to make sure nothing that was already mounted didn't break something else (e.g. keydown events firing in every panel. Solvable, but adds complexity).

It would be great to have maybe a prop to indicate the grid that sizes should be calculated immediately. But if it's a complex change as you say, I understand that maybe the issue should gain a bit more interest from the community.

**Comment by cherniavskii** [comment]
Fixed by https://github.com/mui/mui-x/pull/7155
